cpmpy/transformations/int2bool.py

```diff
@@ -0,0 +1,469 @@
+"""Convert integer linear constraints to pseudo-boolean constraints."""
+
+from typing import List
+import itertools
+import math
+import cpmpy as cp
+from abc import ABC, abstractmethod
+from ..expressions.variables import _BoolVarImpl, _IntVarImpl, boolvar
+from ..expressions.globalconstraints import DirectConstraint
+from ..expressions.core import Comparison, Operator, BoolVal
+from ..expressions.core import Expression
+
+UNKNOWN_COMPARATOR_ERROR = ValueError(
+    "Comparator is not known or should have been simplified by linearize."
+)
+EMPTY_DOMAIN_ERROR = ValueError(
+    "Attempted to encode variable with empty domain (which is unsat)"
+)
+
+
+def int2bool(cpm_lst: List[Expression], ivarmap, encoding="auto"):
+    """Convert integer linear constraints to pseudo-boolean constraints. Requires `linearize` transformation."""
+    assert encoding in (
+        "auto",
+        "direct",
+        "order",
+        "binary",
+    ), "Only auto, direct, order, and binary encoding are supported"
+
+    cpm_out = []
+    for expr in cpm_lst:
+        constraints, domain_constraints = _encode_expr(ivarmap, expr, encoding)
+        cpm_out += domain_constraints + constraints
+    return cpm_out
+
+
+def _encode_expr(ivarmap, expr, encoding):
+    """Return encoded constraints and root-level constraints (e.g. domain constraints exactly-one, ..)."""
+    constraints = []
+    domain_constraints = []
+
+    # skip all Boolean expressions
+    if isinstance(expr, (BoolVal, _BoolVarImpl, DirectConstraint)) or expr.name == "or":
+        return [expr], []
+    elif expr.name == "->":
+        # Encode implication recursively
+        p, consequent = expr.args
+        constraints, domain_constraints = _encode_expr(ivarmap, consequent, encoding)
+        return (
+            [p.implies(constraint) for constraint in constraints],
+            domain_constraints + [p == p],  # keep `p` in model
```

Comment:
I don't understand the p==p part, the p is still in the model because of the p.implies(cons) constraints above?